lib/concurrent/atomic.rb

```diff
@@ -21,7 +22,7 @@
   if /[^0fF]/ =~ ENV['FORCE_ATOMIC_FALLBACK']
     ruby_engine = 'mutex_atomic'
   else
-    ruby_engine = defined?(RUBY_ENGINE)? RUBY_ENGINE : 'ruby'
+    ruby_engine = Concurrent.ruby_engine
```

Comment:
:+1: 

I really like this `ruby_gem` abstraction. We're starting to check this so often (and will continue to do so as we add more optimizations) that we needed a cleaner way to check.